thirdparty/google_analytics/ganalytics.cpp

```diff
@@ -247,11 +247,18 @@ QString GAnalytics::Private::getSystemInfo()
     case QSysInfo::MV_10_11:
         os = "Macintosh; Mac OS 10.11";
         break;
-#if (QT_VERSION >= QT_VERSION_CHECK(5, 8, 0))
     case QSysInfo::MV_10_12:
         os = "Macintosh; Mac OS 10.12";
         break;
-#endif
+    case QSysInfo::MV_10_13:
+        os = "Macintosh; Mac OS 10.13";
+        break;
+    case QSysInfo::MV_10_14:
+        os = "Macintosh; Mac OS 10.14";
+        break;
+    case QSysInfo::MV_10_15:
+        os = "Macintosh; Mac OS 10.15";
+        break;
```

Comment:
Nice try, but these constants do not exist, and so this fails to build on macOS. In fact, the existing `QSysInfo` constants are deprecated. Better to rewrite the function like this:
```
#ifdef Q_OS_MAC
/**
 * Only on Mac OS X
 * Get the Operating system name and version.
 * @return os   The operating system name and version in a string.
 */
QString GAnalytics::Private::getSystemInfo()
{
    auto osType = QOperatingSystemVersion::currentType();
    auto version = QOperatingSystemVersion::current();
    QString os;
    switch (osType)
    {
    case QOperatingSystemVersion::MacOS:
        os = QString("Macintosh; Mac OS ");
        break;
    case QOperatingSystemVersion::IOS:
        os = QString("iPhone; iOS ");
        break;
    default:
        break;
    }
    os += QString("%1.%2").arg(version.majorVersion()).arg(version.minorVersion());
    return os;
}
#endif
```

Reply:
Hmm, I'm pretty sure I looked them up in the Qt 5.15.1 header files?
Will look again...

Reply:
But feel free to PR your proposal ;-)

Reply:
OK, you're right, those don't seem to exist. But your proposal doesn't build either, there seems to be no such thing as a `class QOperatingSystemVersion`.

Reply:
It exists. You just need to `#include <QOperatingSystemVersion>`.

Reply:
As far as I know, there are no CI tests to make sure that MuseScore builds on macOS. Otherwise, it would have failed.

Reply:
Ops, `#include <QOperatingSystemVersion>` helps there...

Reply:
Check #6647